controllers/runner_controller.go

```diff
@@ -872,6 +873,12 @@ func newRunnerPod(template corev1.Pod, runnerSpec v1alpha1.RunnerConfig, default
 		}...)
 	}
 
+	if defaultRunnerImagePullSecretName != "" {
+		pod.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
```

Comment:
I think this should be effective only when `pod.Spec.ImagePullSecrets` is empty(Not sure about RunnerDeployment but at least RunnerSet allows you to specify it via runnerset.spec.template.imagePulSecrets` so I'd like to not break it :)

Reply:
sounds reasonable 👍  updated to be set only if it's not already specified via spec - thx